setuptools/config/expand.py

```diff
@@ -0,0 +1,288 @@
+"""Utility functions to expand configuration directives or special values
+(such glob patterns).
+
+We can split the process of interpreting configuration files into 2 steps:
+
+1. The parsing the file contents from strings to value objects
+   that can be understand by Python (for example a string with a comma
+   separated list of keywords into an actual Python list of strings).
+
+2. The expansion (or post-processing) of these values according to the
+   semantics ``setuptools`` assign to them (for example a configuration field
+   with the ``file:`` directive should be expanded from a list of file paths to
+   a single string with the contents of those files concatenated)
+
+This module focus on the second step, and therefore allow sharing the expansion
+functions among several configuration file formats.
+"""
+import ast
+import importlib
+import io
+import os
+import sys
+from glob import iglob
+from itertools import chain
+
+from distutils.errors import DistutilsOptionError
+
+chain_iter = chain.from_iterable
+
+
+class StaticModule:
+    """
+    Attempt to load the module by the name
```

Comment:
This feels wrong. This verb at the beginning suggests that this thing the docstring describes is callable (function/method or has `__call__()`). But it isn't. Maybe consider describing what this object is? And move the side-effect description to where it happens, like the initializer docstring.

Reply:
Thank you very much @webknjaz. This comes from the [original implementation](https://github.com/pypa/setuptools/blob/fe7975d091483e8d6d4c9befea38ccfeb29e8e49/setuptools/config.py#L22) that was moved/extracted.

I will try to think what would be more appropriate.